fluvii/kafka_tools/fluvii_toolbox.py

```diff
@@ -92,51 +93,107 @@ def create_topics(self, topic_config_dict, ignore_existing_topics=True):
         for topic in topic_config_dict:
-            current_configs[topic].update(topic_config_dict[topic])
-            topic_config_dict[topic] = current_configs[topic]
-        if topic_config_dict:
-            futures = self.admin.alter_configs([ConfigResource(2, topic, set_config=configs) for topic, configs in topic_config_dict.items()])
-            for topic in topic_config_dict:
-                futures[topic].result()
-        LOGGER.info(f'Altered topics: {list(topic_config_dict.keys())}')
+            config_updates[topic].update(topic_config_dict[topic])
+        altered = []
+        futures = self.admin.alter_configs([ConfigResource(2, topic, set_config=config_updates[topic]) for topic in topic_config_dict])
+        try:
+            for future_name, future in futures.items():
+                future.result()
+                topic_config_dict.pop(future_name.name)
+                altered.append(future_name.name)
+        except KafkaException as e:
+            if e.args[0].name() == 'POLICY_VIOLATION':
+                bad_fields = e.args[0].str().split(":")[2:]
+                unhandled_errors = [field for field in bad_fields if 'This config cannot be updated.' not in field]
+                uneditable = [field.split('=')[0].strip() for field in bad_fields if 'This config cannot be updated.' in field]
+                if unhandled_errors and not uneditable:
+                    raise ValueError(unhandled_errors)
+                ignore_field_list = list(ignore_field_list) + uneditable
+                LOGGER.info(f'Restricted configs were discovered (which may be from ensuring all previous set configs remain): {uneditable}. '
+                            f'Retrying with them added to the ignore field list. If you do not see a field you directly provided here, all should be well!')
+                LOGGER.info(f'Topics altered successfully so far: {altered}')
+                return self.alter_topics(topic_config_dict, retain_configs=retain_configs, ignore_missing_topics=ignore_missing_topics, ignore_field_list=ignore_field_list)
+            else:
+                raise
+        LOGGER.info(f'Topics altered successfully: {altered}')
 
     def delete_topics(self, topics, ignore_missing=True):
         self._set_admin()
+        topics = set(topics)
         if ignore_missing:
             existing = set(self.list_topics())
-            missing = set(topics) - existing
+            missing = topics - existing
             if missing:
                 LOGGER.info(f'These topics dont exist, ignoring: {missing}')
                 topics = [i for i in topics if i not in missing]
         if topics:
+            topics = list(topics)
             futures = self.admin.delete_topics(topics)
             for topic in topics:
                 futures[topic].result()
         LOGGER.info(f'Deleted topics: {topics}')
 
+    def sync_topics(self, topic_config_dict):
```

Comment:
Consider adding logging statements to the `sync_topics` method to provide more visibility into what it is doing and whether it succeeds or fails. This can be useful for troubleshooting and monitoring.